src/tools/SimpleTraversability.hpp

```diff
@@ -0,0 +1,192 @@
+            : maximumSlope(0)
+            , classCount(0)
+            , groundClearance(0)
+            , minPassageWidth(0)
+            , obstacleClearance(0)
+        {
+        }
+
+        SimpleTraversabilityConfig(
+                double maximumSlope,
+                int classCount,
+                double groundClearance,
+                double minPassageWidth = 0,
+                double obstacleClearance = 0)
+
+            : maximumSlope(maximumSlope)
+            , classCount(classCount)
+            , groundClearance(groundClearance)
+            , minPassageWidth(minPassageWidth)
+            , obstacleClearance(obstacleClearance)
+        {
+        }
+
+    };
+
+    /**
+     * @brief Classification of terrain into symbolic traversability classes, based on
+     * geometric constraints
+     *
+     * For now, the modalities that are used are:
+     *
+     * <ul>
+     * <li>maximum slope
+     * <li>maximum step size
+     * </ul>
+     *
+     * The (very simple) model maps the local slope from [@a maximumSlope, 1] to [0, 1].
+     * The resulting value is then quantified in @a classCount intervals.
+     * Probability will be set to 1 for every classification beside CLASS_UNKNOWN.
+     *
+     * It outputs a TraversabilityGrid in which each cell has an integer value,
+     * this integer value being the traversability class for the cell. Since
+     * CLASS_UNKNOWN (=0) and CLASS_OBSTACLE (=1) are reserved, the resulting
+     * traversability class is in [2, 2 + classCount - 1]
+     *
+     * If one of the modalities is missing, it is simply ignored
+     */
+    class SimpleTraversability {
+
+        enum INPUT_DATA {
+            SLOPE,
+            MAX_STEP,
+            INPUT_COUNT
+        };
+
+    public:
+        enum CLASSES {
+            CLASS_UNKNOWN = 0,
+            CLASS_OBSTACLE = 1,
+            CUSTOM_CLASSES = 2
+        };
+    private:
+
+        SimpleTraversabilityConfig config;
+
+    public:
+        SimpleTraversability();
+
+        SimpleTraversability(
+                const SimpleTraversabilityConfig& config);
+
+        SimpleTraversability(
+                double maximumSlope,
+                unsigned int classCount,
+                double groundClearance,
+                double minPassageWidth = 0,
+                double obstacleClearance = 0);
+
+        void setConfig(const SimpleTraversabilityConfig& config);
+        void setConfig(
+                double maximumSlope,
+                unsigned int classCount,
+                double groundClearance,
+                double minPassageWidth = 0,
+                double obstacleClearance = 0);
+
+        const SimpleTraversabilityConfig& getConfig() const;
+
+        /**
+         * Calculates the traversability based on the input maps.
+         * @details:
+         * Note that the default value of the input maps will be
+         * used to determine UNKNOWN cells.
+         */
+        bool calculateTraversability(
```

Comment:
calculateTraversability, closeNarrowPassages and growObstacles could be const right?

Reply:
Yes.